kr8s/tests/test_config.py

```diff
@@ -51,7 +51,14 @@ async def test_kubeconfig_from_dict(temp_kubeconfig, cls):
     with open(temp_kubeconfig) as fh:
         config = yaml.safe_load(fh)
     kubeconfig = await cls(config)
-    assert kubeconfig.raw == config
+    assert kubeconfig.raw["clusters"] == config["clusters"]
+    assert kubeconfig.raw["users"] == config["users"]
+    assert kubeconfig.raw["contexts"] == config["contexts"]
+    # Preferences and extensions are optional
+    if "preferences" in config:
+        assert kubeconfig.raw["preferences"] == config["preferences"]
```

Comment:
Wouldn't it make sense to compare like this (without if statement)
```python
assert kubeconfig.raw.get("preferences", {}) == config.get("preferences", {})
```

Reply:
Yeah that's definitely more elegant.